lib/foreman_providers/compute_resource.rb

```diff
@@ -0,0 +1,35 @@
+module ForemanProviders
+  module ComputeResource
+    def self.included(base)
+      base.class_eval do
+        after_create   :create_provider
+        before_destroy :destroy_provider
+
+        private
+
+        def create_provider
+          provider_klass = foreman_type_to_provider_type.constantize
+          provider_klass.create!(
+            :name     => name,
+            :hostname => URI(url).host,
+          ).tap do |ems|
+            ems.update_authentication(:default => {:userid => user, :password => password})
```

Comment:
this means we store the credential twice? you would need to handle an update to U/P as well. ideally, we should just use the CR credentials? (or if providers has a better way of storing it, replace the current way CR does it).

Reply:
Correct eventually we would move to using a single credential store, the MIQ Authentications model supports a wide array of auth types and allows for multiple credentials for e.g. different endpoints under the same provider.  Ovirt is a good example where metrics collection requires different credentials from the main inventory endpoint